### PR TITLE
Added  SolarPrediction API as alternative service

### PR DESCRIPTION
**Pull request by zoernert**
- fixed linting issues
- added api service

### CODE REVIEW

File: admin/jsonConfig.json [hunk 1]
```diff
@@ -21,7 +25,7 @@
                             "value": "solcast"
                         }
                     ],
-                    "default": "forecastsolar",
+                    "default": "spa",
```
Comment (klein0r) [inline admin/jsonConfig.json:28]: Why changed default?

File: io-package.json [hunk 2]
```diff
@@ -1,8 +1,11 @@
 {
   "common": {
     "name": "pvforecast",
-    "version": "2.3.0",
+    "version": "2.3.2",
```
Comment (klein0r) [inline io-package.json:4]: Please do not update the version manually - the release script will do that.

File: io-package.json [hunk 3]
```diff
@@ -1,8 +1,11 @@
...
     "news": {
+      "2.3.1": {
```
Comment (klein0r) [inline io-package.json:6]: News will be generated automatically (in all languages) by the release script (source is README.md).
Just remove this part. At the moment the version here doesn't match the io-package.json

File: main.js [hunk 4]
```diff
@@ -546,8 +548,15 @@ class Pvforecast extends utils.Adapter {
 				}
 			} else if (this.config.service === 'solcast') {
 				url = `https://api.solcast.com.au/world_pv_power/forecasts?format=json&hours=48&loss_factor=1&latitude=${this.latitude}&longitude=${this.longitude}&tilt=${plant.tilt}&azimuth=${this.convertAzimuth(plant.azimuth)}&capacity=${plant.peakpower}&api_key=${this.config.apiKey}`;
+			} else if (this.config.service === 'spa') {
+				url = `https://solarenergyprediction.p.rapidapi.com/v2.0/solar/prediction?decoration=forecast.solar&lat=${this.latitude}&lon=${this.longitude}&deg=${plant.tilt}&az=${plant.azimuth}&kwp=${plant.peakpower}`;
```
Comment (klein0r) [inline main.js:552]: If the API key is mandatory for this service, add some checks to onReady

File: package.json [hunk 5]
```diff
@@ -1,12 +1,16 @@
 {
   "name": "iobroker.pvforecast",
-  "version": "2.3.0",
+  "version": "2.3.1",
```
Comment (klein0r) [inline package.json:3]: Please do not update the version manually - the release script will do that.
This is the best example why you shouldn't change it manually: At the moment this version doesn't match io-package.json

File: admin/jsonConfig.json (hunk 1 above)
Comment (zoernert) [reply admin/jsonConfig.json:28]: Sorry was to fasten my development. Will revert in another commit.

**Pull request closed without merging**